website/static/js/home-page/donateBannerPlugin.js

```diff
@@ -0,0 +1,65 @@
+var m = require('mithril');
+var $osf = require('js/osfHelpers');
+
+// CSS
+require('css/donate-banner.css');
+require('css/meetings-and-conferences.css');
+
+var currentDate = new Date();
+
+//TODO: Change these dates when we know
+var changeDates = [
+    new Date('July 18, 2017'),
+    new Date('July 19, 2017'),
+    new Date('July 20, 2017'),
+    new Date('July 21, 2017')
+];
+
+var bannerText = 'The Center for Open Science (COS) created the OSF and a suite of of free products to advance the ' +
+    'work of the research community. If you value these tools, please make a gift to support COS’s efforts to improve' +
+    ' and scale these services.';
+
+var background = '.donate-banner';
+
+if (currentDate > changeDates[0]){
```

Comment:
This loop is never going to get past the first date, since the dates are ascending and you check the earliest one first. Lean towards `currentDate >= changeDates[x] and currentDate < changeDates[x+1]` for these, and have the last date be the ending date of the whole campaign where there won't be a banner at all.